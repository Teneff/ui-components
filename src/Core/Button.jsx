import React from 'react';
import PropTypes from 'prop-types';

export default function Button({ text }) {
  return (
    <button
      type="button"
      className="btn btn-primary btn-default  more-info-btn chevron"
    >
      {text}
    </button>
  );
}

Button.propTypes = {
  text: PropTypes.string.isRequired,
};
