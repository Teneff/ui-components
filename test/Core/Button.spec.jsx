import React from 'react';
import { shallow } from 'enzyme';
import Button from '../../components/Core/Button';

describe('<Button />', () => {
  const component = shallow(<Button text="Some Text" />);

  it('should render a button with text', () => {
    expect(component).toMatchSnapshot();
  });
});
