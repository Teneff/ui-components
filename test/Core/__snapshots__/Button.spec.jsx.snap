// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render a button with text 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    text="Some Text"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Some Text",
      "className": "btn btn-primary btn-default  more-info-btn chevron",
      "type": "button",
    },
    "ref": null,
    "rendered": "Some Text",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Some Text",
        "className": "btn btn-primary btn-default  more-info-btn chevron",
        "type": "button",
      },
      "ref": null,
      "rendered": "Some Text",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
